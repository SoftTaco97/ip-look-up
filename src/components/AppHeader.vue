<template>
  <v-container fluid>
    <v-row justify="center">
      <v-col cols="12">
        <h2>IP Address Look Up</h2>
      </v-col>
    </v-row>
    <search-form
      :search="ipAddress"
      @update-search="updateIpAddress"
      @submit="requestIpData"
    />
  </v-container>
</template>

<script>
import SearchForm from './SearchForm.vue';

export default {
  name: 'AppHeader',
  props: {
    ipAddress: {
      type: String,
      required: true,
    },
  },
  methods: {
    updateIpAddress(val) {
      this.$emit('update-search', val);
    },
    requestIpData() {
      this.$emit('submit');
    },
  },
  components: { SearchForm },
};
</script>
