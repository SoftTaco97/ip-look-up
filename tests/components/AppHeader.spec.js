import createComponentMounter from '../componentMounter';
import createStubs from '../createStubs';
import AppHeader from '../../src/components/AppHeader.vue';

describe('src/components/AppHeader.vue unit tests', () => {
  const stubs = createStubs(['SearchForm']);
  const mountComponent = createComponentMounter(AppHeader, {
    stubs,
    propsData: {
      loading: false,
      userSearch: '',
    },
  });
  let wrapper;

  beforeEach(() => {
    wrapper = mountComponent();
  });

  afterEach(() => wrapper.destroy());

  describe('Rendering unit tests', () => {
    it('Should match the snapshot', () => {
      // ! - Deep mount for snapshots
      wrapper = mountComponent({}, false);
      expect(wrapper).toMatchSnapshot();
    });
    it('Should have a main title', () => {
      expect(wrapper.html()).toEqual(expect.stringContaining('IP Address Look Up'));
    });
  });
  describe('Method unit tests', () => {
    describe('updateSearch unit tests', () => {
      it('Should emit the update-search event with the value provided', () => {
        wrapper.vm.updateSearch('test');
        expect(wrapper.emitted()['update-search']).toBeTruthy();
        expect(wrapper.emitted()['update-search'][0]).toEqual(['test']);
      });
    });
    describe('requestIpData unit tests', () => {
      it('Should emit the submit event', () => {
        wrapper.vm.requestIpData();
        expect(wrapper.emitted().submit).toBeTruthy();
      });
    });
  });
});
