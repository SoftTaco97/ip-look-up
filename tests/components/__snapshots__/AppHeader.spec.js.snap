// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/AppHeader.vue unit tests Rendering unit tests Should match the snapshot 1`] = `
<div class="container container--fluid">
  <div class="row justify-center">
    <div class="col col-12">
      <h2>IP Address Look Up</h2>
    </div>
  </div>
  <div search=""></div>
</div>
`;
